pkg/parsers/operatingsystem/operatingsystem_linux.go

```diff
@@ -18,15 +23,29 @@ var (
 
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
-	b, err := ioutil.ReadFile(etcOsRelease)
+	osReleaseFile, err := os.Open(etcOsRelease)
 	if err != nil {
 		return "", err
 	}
-	if i := bytes.Index(b, []byte("PRETTY_NAME")); i >= 0 {
-		b = b[i+13:]
-		return string(b[:bytes.IndexByte(b, '"')]), nil
+	defer osReleaseFile.Close()
+
+	scanner := bufio.NewScanner(osReleaseFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "PRETTY_NAME") {
+			data := strings.SplitN(line, "=", 2)
+			if len(data) == 2 {
+				prettyNames, err := shellwords.Parse(data[1])
+				if err == nil && len(prettyNames) == 1 {
+					return prettyNames[0], nil
+				}
```

Comment:
I think we should return "err" here so that people know why we didn't like the value.